### DISCUSSION TITLE
Alternative to the "formula" Column

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#6092] [Help]
I am hoping someone can help me

I am creating an app to record the work that the company has carried out and allow customers to log on to view the work that we have done for them.

Currently I am using the `formula` column option but it appears that this does not suit the purpose that we need for a few reasons:

- these cells cannot be called via SQL queries
- I cannot use these cells in a spin off app where I call the same table
- ...

I was wondering if there was another way to achieve the same result but by using a standard column type?

one of the formula cells, for example, is used to calculate the job profit and takes the values for around 5 columns to calculate via a JS script the job profit.

I have tried using a number cell and using the same JS script in the cell on the page but when I save the entry, the cell does not contain the calculation, it is left blank.

Is there a work around that someone could share?

My main issue is the SQL queries as some of my customers like visual representations and the ability to show these figures in a graph would be excellent

**Comment by melohagan** [comment]
Hi @PaulMcF1987 

I agree with your summation that the Formula field type isn't going to cut it for this use case.

>I have tried using a number cell and using the same JS script in the cell on the page but when I save the entry, the cell does not contain the calculation, it is left blank.

Could you share the JavaScript binding that you are using alongside your component tree please? Or I can have a look in your workspace if it's up-to-date and you provide the field and screen names.

**Reply by PaulMcF1987** [reply]
Hi @melohagan 

I do not have the details of the binding that was used as I tried a few different things and eventually gave up.

Is there a recomended way to achieve the desired outcome?

**Comment by mike12345567** [comment]
Hi @PaulMcF1987 - I've converted this to a help discussion.

Right now, formulas are calculated in Budibase, when the data is retrieved, meaning that the column isn't actually written to your SQL table.

You have two options I can think of:
1. Create a column for the resulting type, e.g. if its a numeric calculation then create an integer column in your SQL table, then save it as part of the form write, as a calculated column.
2. Depending on the type of SQL database you are using, you can try using a computed column.

**Reply by PaulMcF1987** [reply]
@mike12345567 I have made a computed cell in my SQL database using HeidiSQL. Under Vurtuality I chose stored.

I then, from the budibase application, chose `fetch tables` but for some reason it is not displaying the new fields in Budibase...

I deleted the columns in HeidiSQL thinking that if I add them on Budibase and then add the Formula on HeidiSQL that it might then work, however, when I try to add the column I get an error saying `column name already in use` however the column name is not in use as it has been deleted
I have tried using the fetch tables option and then trying again but no joy.

I changed the column name and now I get a conflict on every table etc telling me that the there is an unknown column

**Reply by PaulMcF1987** [reply]
When I run an SQL query `select * from table` It shows me all the columns/rows that I would expect. 
![image](https://user-images.githubusercontent.com/75368133/170748746-e33e011e-ce65-4416-a89e-26f011f6d339.png)

However, after fetching the tables, I currently have:
![image](https://user-images.githubusercontent.com/75368133/170748210-49185586-0661-4ae2-a37d-6d5b954b1fdc.png)

There must be some sort of issue with Budibase fetching the table...

I could work off the SQL query, however, the relationship set up means that there will be a number of fields that will not be available to me via the SQL query.

When I go into the design area, I cannot do anything because there is an error at the top of the screen telling me that there is an unrecognised table column.

**Comment by PaulMcF1987** [comment]
@melohagan  @mike12345567 
Is someone able to explain why tables disappear?

The pic shared above showing the jobs table with nothing in it despite having 3 test jobs added which are still showing when I view Heidi SQL...
The same thing has happened with all of the tables which have been joined via the relationships, they are all now showing as blank

**Reply by mike12345567** [reply]
Hi @PaulMcF1987 - it sounds like there may be a bug here, if you can provide the type of SQL database you're using and the definition of the computed column we will take a look at it and see if we can recreate.

If you have removed the column and it appears to still be presenting as an issue, you may need to delete the datasource and re-import it, or rename the table as a temporary fix, the other option is to wait for the fix for the table import if we can recreate.
